hollow/src/test/java/com/netflix/hollow/api/consumer/index/HashIndexTest.java

```diff
@@ -57,7 +58,11 @@ public static void setup() {
 
             for (int i = 0; i < 100; i++) {
                 ws.add(new DataModel.Producer.TypeA(1, "TypeA" + i));
+                ws.add(new DataModel.Producer.TypeB(1, "TypeB" + i));
```

Comment:
Something odd is going  on - somehow adding there here causes TestManyMatches#testWithSelectGenericHollowObject,testWithSelect to fail. Digging further...

Reply:
I can reproduce using an old-style manually constructed HollowHashIndex. It only happens when I pass in a select field path for the string and I _don't_ explicitly as for `.value` as the select field path. This is explicitly disallowed in the new-style hash indexes - need to sync up with Paul on this.

Reply:
The field path parsing should be consistent with `com.netflix.hollow.core.index.HollowHashIndex` and `com.netflix.hollow.api.consumer.index.HashIndex`. See use of `FieldPaths.createFieldPathForHashIndex`.  Auto-expansion is not supported for the select and match field paths of a hash index.

Reply:
I found the cause.  Select path never quite worked correctly, it currently requires the path have a last segment reference a field value, and the ordinal returned will correspond to the enclosing record.

The problematic line is at https://github.com/Netflix/hollow/blob/master/hollow/src/main/java/com/netflix/hollow/core/index/traversal/TraversalTreeBuilder.java#L103:
```java
                            if(schema.getFieldType(fieldSchemaPositions[i]) == FieldType.REFERENCE) {
                                fieldSchemaPositions[i] = -1;
                                fieldTypeDataAccess[i] = getChildDataAccess(typeDataAccess, pathElement);
                            } else {
                                fieldSchemaPositions[i] = schema.getPosition(pathElement);
                                fieldTypeDataAccess[i] = typeDataAccess;
                            }
```

It should be:
```java
                            if(schema.getFieldType(pathElement) == FieldType.REFERENCE) {
```

Reply:
nice - great fine!